packages/algob/src/lib/tx.ts

```diff
@@ -248,3 +251,54 @@ export async function transferASALsig (
 
   return await deployer.waitForConfirmation(tx1.txId);
 }
+
+/**
+ * Description:
+ * This function executes multiple transactions atomically where Algos are
+ * transferred from one account to another using logic signature (for signing)
+
+ * Returns:
+ * Transaction details
+*/
+
+export async function transferMicroAlgosLsigAtomic (
+  deployer: AlgobDeployer,
+  grpTxnParams: GrpTxnParams[]
+): Promise<tx.ConfirmedTxInfo> {
+  if (grpTxnParams.length > 16) {
+    throw new Error("Maximum size of an atomic transfer group is 16");
+  }
+
+  const txns = [];
+  for (const p of grpTxnParams) {
+    const params = await mkSuggestedParams(deployer.algodClient, p.payFlags);
+    const receiver = p.toAccountAddr;
+    const note = encodeNote(p.payFlags.note, p.payFlags.noteb64);
+
+    const txn = tx.makePaymentTxnWithSuggestedParams(
+      p.fromAccount.addr, receiver, p.amountMicroAlgos, p.payFlags.closeRemainderTo, note, params);
+
+    txns.push(txn); // group transactions
+  }
+
+  tx.assignGroupID(txns); // assign common group hash to all transactions
+
+  const signed = []; const logs = []; let idx = 0;
+  for (const txn of txns) {
+    const signedTxn = tx.signLogicSigTransactionObject(txn, grpTxnParams[idx].lsig);
+    signed.push(signedTxn.blob);
+
+    const txnInfo = {
+      from: grpTxnParams[idx].fromAccount.addr,
+      to: grpTxnParams[idx].toAccountAddr,
+      amount: txn.amount,
+      txid: signedTxn.txID
+    };
+    logs.push(txnInfo);
+    idx++;
+  }
+
+  const pendingTx = await deployer.algodClient.sendRawTransaction(signed).do();
+  console.log(ALGO_MSG, logs);
+  return await deployer.waitForConfirmation(pendingTx.txId);
+}
```

Comment:
I agree with @amityadav0  - this function is too specialized. Probably it's better to move it to the example `common.js`  file (unless we need it in other place).